dashboard/app/helpers/users_helper.rb

```diff
@@ -12,6 +12,29 @@ module UsersHelper
   ACCT_TAKEOVER_OAUTH_TOKEN = 'clever_takeover_token'
   ACCT_TAKEOVER_FORCE_TAKEOVER = 'force_clever_takeover'
 
+  # Move followed sections from old_user to new_user and destroy old_user
+  def move_sections_and_destroy_old_user(old_user, new_user)
+    # No-op if old_user is nil
+    return true unless old_user.present?
+
+    if old_user.has_activity?
+      # We don't want to destroy an account with progress.
+      # In theory this should not happen, so we log a Honeybadger error and return.
+      Honeybadger.notify(
+        error_class: 'Oauth takeover called for user with progress',
+        error_message: "Attempted to take over account with id #{old_user.id}, which has activity"
+      )
+      return false
+    end
+
+    # Move over sections that old_user follows
+    if new_user.student?
+      Follower.where(student_user_id: old_user.id).update_all(student_user_id: new_user.id)
+    end
+
+    old_user.destroy! ? true : false
```

Comment:
To consider: Should we wrap these two update operations in a transaction?  Or are we already considering a case where we successfully move the followers to `new_user` but fail to destroy `old_user`?

Reply:
i was struggling with that. there isn't a way that i know of to tell if `update_all` is successful since it bypasses validations and only throws (which would cause the transaction to rollback) if we mess something up at the database level and get an [ActiveRecord::StatementInvalid error](https://stackoverflow.com/questions/36971798/what-happens-when-update-all-fails). 

i also like that `update_all` is more performant than calling `update!` on each record, but i suppose that there aren't enough followed sections for a student where that would really matter a ton?

so this could be refactored to:
```
transaction do
  Follower.where(student_user_id: old_user.id).each { |followed| followed.update!(student_user_id: new_user.id) }
  old_user.destroy!
end
```